examples/async/fastapi_example.py

```diff
@@ -1,13 +1,42 @@
+from contextlib import asynccontextmanager
+
 import async_module
 import fastapi
+import hamilton_sdk.adapters
 
 from hamilton import base
 from hamilton.experimental import h_async
 
-app = fastapi.FastAPI()
-
+tracker_async = hamilton_sdk.adapters.AsyncHamiltonTracker(
+    project_id=4,
+    username="elijah",
+    dag_name="sync_tracker",
+)
+# tracker_sync = hamilton_sdk.adapters.HamiltonTracker(
+#     project_id=4,
+#     username="elijah",
+#     dag_name="sync_tracker",
+# )
 # can instantiate a driver once for the life of the app:
-dr = h_async.AsyncDriver({}, async_module, result_builder=base.DictResult())
+dr = None
+
+
+@asynccontextmanager
+async def lifespan(app: fastapi.FastAPI):
+    global dr
+    dr = await h_async.AsyncDriver(
+        {},
+        async_module,
+        result_builder=base.DictResult(),
+        adapters=[
+            tracker_async,
+            # tracker_sync
+        ],
+    ).ainit()
```

Comment:
ideally we also add a builder to make this match the new way we're doing things?

Reply:
See TODO